data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/export/DataFileScheduler.java

```diff
@@ -133,23 +139,39 @@ public void shutdown() {
     }
 
     private void processDataFilePartition(DataFilePartition dataFilePartition) {
+        // Create AcknowledgmentSet
+        final boolean isAcknowledgmentsEnabled = sourceConfig.isAcknowledgmentsEnabled();
+        AcknowledgementSet acknowledgementSet = null;
+        if (sourceConfig.isAcknowledgmentsEnabled()) {
+            acknowledgementSet = acknowledgementSetManager.create((result) -> {
+                if (result) {
+                    completeDataLoader(dataFilePartition).accept(null, null);
+                    LOG.info("Received acknowledgment of completion from sink for data file {}", dataFilePartition.getKey());
+                } else {
+                    exportFileErrorCounter.increment();
+                    LOG.warn("Negative acknowledgment received for data file {}, retrying", dataFilePartition.getKey());
+                    sourceCoordinator.giveUpPartition(dataFilePartition);
```

Comment:
I would assume the retry on this partition and also keeping track of retryCount is handled by the sourceCoordinator?

Reply:
Yes. Once the datafile partition is given up, it will be available for acquire again and the file load will be retried by whichever node that picked up the partition.